### ISSUE TITLE
Details on fine-tuning data

### ISSUE THREAD

**Issue by phlobo** [cambridgeltl/sapbert#4]
Hello and thanks for sharing this great project.

Regarding fine-tuning of SapBERT, the README states

> For finetuning on your customised dataset, generate data in the format of [...]
where entity_name_1 and entity_name_2 are synonym pairs (belonging to the same concept concept_id) sampled from a given labelled dataset. 

Are there any examples on how this looks exactly for the datasets (NCBI Disease, Cometa, etc.) used in the evaluation?

**Comment by hardyqr** [comment]
Hi, apologies for the long-delayed reply. Have you checked out the readme for evaluation: https://github.com/cambridgeltl/sapbert/tree/main/evaluation ?
If you follow the link to download the evaluation datasets there will be data formats for both the target ontology and also the queries.

**Comment by phlobo** [comment]
Thank you for your response! 
However, it looks like the generated synonym pairs for fine-tuning on a specific dataset are not part of the zip files?

Two questions I have:
- Do only fine-tune on pairs of mentions from the training part of a specific dataset, or do you also take into account synonyms? 
- What happens if a specific CUI has only one surface form in the data? Will this CUI be excluded from fine-tuning?

**Comment by hardyqr** [comment]
Yes indeed. Those data are just for testing but not training/finetuning.

> Do only fine-tune on pairs of mentions from the training part of a specific dataset, or do you also take into account synonyms?

In my original paper, it should only contain the training part of a specific dataset. I guess one could also take into account all synonyms extracted from an ontology. That's not something I tried but I imagine it could work quite well!

> What happens if a specific CUI has only one surface form in the data? Will this CUI be excluded from fine-tuning?
Good point. In this case the CUI would be excluded from finetuning.

I couldn't find the original finetuning files I generated anymore. But in principle, what I did was:
Given a training file containing synonym list in the format of 
```
CUI_1 || AAA
CUI_1 || BBB
CUI_2 || CCC
CUI_3 || DDD
CUI_3 || EEE
CUI_3 || FFF
```
I generate all pairwise enumerations of mentions belonging to the same CUI.  For the toy example shown above, that list will be:
```
CUI_1 || AAA || BBB
CUI_3 || DDD || EEE
CUI_3 || DDD || FFF
CUI_3 || DDD || EEE
```

I hope this helps.
